seqr/views/apis/staff_api_2_3_tests.py

```diff
@@ -676,16 +679,18 @@ def test_sample_metadata_export(self):
             response.json()['message'],
             'Found multiple airtable records for sample NA19675 with mismatched values in field dbgap_study_id')
         self.assertEqual(len(responses.calls), 2)
-        self.assertIsNone(responses.calls[0].request.params.get('offset'))
-        self.assertEqual(responses.calls[1].request.params.get('offset'), 'abc123')
+        queries = {item[0]: item[1] for item in [query.split('=') for query in utils.urlparse(responses.calls[0].request.url).query.split('&')]}
```

Comment:
why did this change?

Reply:
It seems that the `responses` in Python 3 couldn't always parse the parameters in the `url`  correctly. 
The error report as below:
```
======================================================================
FAIL: test_sample_metadata_export (seqr.views.apis.staff_api_2_3_tests.StaffAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<string>", line 3, in wrapper
  File "/Users/shifa/dev/seqr/seqr/views/apis/staff_api_2_3_tests.py", line 683, in test_sample_metadata_export
    self.assertEqual(responses.calls[1].request.params.get('offset'), 'abc123')
AssertionError: None != 'abc123'

----------------------------------------------------------------------
Ran 175 tests in 14.821s

FAILED (failures=1)
Destroying test database for alias 'default' ('test_seqrdb')...
Destroying test database for alias 'reference_data' ('test_reference_data_db')...
(seqr) wm598-921:seqr shifa$ pip list|grep responses
responses                            0.10.14   
```
When I check the value of `responses.calls[1].request.url`, it contains `?offset=abc123` and it is the same as that in Python 2. Therefore, I wrote a line of code to parse the url and got the value for `offset`.

Reply:
Its really concerning to me that the call arguments changed, and we should definitely not be parsing the query string like this when we use the responses package. Can you look into the `responses.calls[0].request`  more in both python versions and figure out whats different and why?

Reply:
While comparing the params of the requests, we can see the differences in the params:
Python 2:
```
u'responses.calls[1].request.params': {'offset': 'abc123', 'filterByFormula': "OR({CollaboratorSampleID}='NA20885',{SeqrCollaboratorSampleID}='NA20885',{CollaboratorSampleID}='NA20888',{SeqrCollaboratorSampleID}='NA20888',{CollaboratorSampleID}='NA20889',{SeqrCollaboratorSampleID}='NA20889')", 'fields[]': 'dbgap_submission'}
```
Python 3:
```
'responses.calls[1].request.params': {'/v0/app3Y97xtbbaOopVR/Samples?offset': 'abc123', 'fields[]': 'dbgap_submission', 'filterByFormula': "OR({CollaboratorSampleID}='NA20888',{SeqrCollaboratorSampleID}='NA20888',{CollaboratorSampleID}='NA20889',{SeqrCollaboratorSampleID}='NA20889',{CollaboratorSampleID}='NA20885',{SeqrCollaboratorSampleID}='NA20885')"}

```

Reply:
so then we should still be using the `params` just find the key that contains `offset` instead of the one that exactly equals it, but theres no reason to url parse the whole thing ourselves

Reply:
I upgraded the version of `responses` to `0.10.15`. Then, the issue disappears.